src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/exception/CommunityApiCallError.kt

```diff
@@ -0,0 +1,75 @@
+6) _ Conflict from POST https://community-api-secure.probation.service.justice.gov.uk/secure/offenders/crn/_/appointments/_/reschedule/context/commissioned-rehabilitation-services*/
+
+class CommunityApiCallError(val httpStatus: HttpStatus, causeMessage: String, val responseBody: String, val exception: Throwable) :
+  RuntimeException(
+    exception
+  ) {
+  val category = errorCategoryByRemovingIdentifiers(causeMessage)
+  private val error = when {
+    httpStatus.is4xxClientError -> {
+      when {
+        category.contains("Contact type .* requires an outcome type as the contact date is in the past".toRegex()) -> {
+          Pair("Delius requires an appointment to only be made for today or in the future. Please amend and try again", true)
+        }
+        category.contains("endTime must be after or equal to startTime".toRegex()) -> {
+          Pair("Delius requires that an appointment must end on the same day it starts. Please amend and try again", true)
+        }
+        category.contains("CRN: _ EventId: _ has multiple referral requirements".toRegex()) -> {
+          Pair("The Service User Sentence must not contain more than one active rehabilitation activity requirement. Please correct in Delius and try again", true)
+        }
+        category.contains("Cannot find NSI for CRN: _ Sentence: _ and ContractType".toRegex()) -> {
+          Pair("There has been an error during creating the Delius NSI. We are aware of this issue. For follow-up, please contact support", true)
+        }
+        category.contains("Multiple existing matching NSIs found".toRegex()) -> {
+          Pair("There has been an error during creating the Delius NSI. We are aware of this issue. For follow-up, please contact support", true)
+        }
+        category.contains("Conflict.*appointments".toRegex()) -> {
+          Pair("The appointment conflicts with another. Please contact the service user's probation practitioner for an available slot and try again", true)
+        }
+        category.contains("Conflict.*reschedule".toRegex()) -> {
+          Pair("The appointment conflicts with another. Please contact the service user's probation practitioner for an available slot and try again", true)
+        }
+        else -> {
+          Pair("Delius reported \"$category\". Please correct, if possible, otherwise contact support", true)
+        }
+      }
+    }
+    else -> {
+      Pair("System is experiencing issues. Please try again later and if the issue persists contact Support", true)
+    }
+  }
+  val userMessage = error.first
+  val logError = error.second
+
+  private fun errorCategoryByRemovingIdentifiers(message: String): String {
+    val messageWithoutIds = "[A-Z]*[0-9]+".toRegex().replace(message, "_")
+    return "'[A-Z0-9]*'".toRegex().replace(messageWithoutIds, "'_'")
```

Comment:
`'CRSAA'` is probably important to keep for us

Reply:
Changed to only replace numerics with _